pkg/k8sclient/kubeconfig.go

```diff
@@ -87,8 +89,23 @@ func PerNodeK8sClient(nodeName, bootstrapKubeconfigFile string, certDuration tim
 	// https://github.com/kubernetes/kubernetes/blob/068ee321bc7bfe1c2cefb87fb4d9e5deea84fbc8/cmd/kubelet/app/server.go#L953-L963
 	newClientsetFn := func(current *tls.Certificate) (kubernetes.Interface, error) {
 		cfg := bootstrapKubeconfig
-		if current != nil {
-			cfg = config
+
+		// validate the kubeconfig
+		tempClient, err := kubernetes.NewForConfig(cfg)
+		if err != nil {
+			logging.Errorf("failed to read kubeconfig from cert manager: %v", err)
+		} else {
+			_, err := tempClient.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+			// tls unknown authority error is unrecoverable error with retry
+			if err != nil && strings.Contains(err.Error(), "x509: certificate signed by unknown authority") {
+				// reload and use bootstrapKubeconfig again
+				newBootstrapKubeconfig, _ := clientcmd.BuildConfigFromFlags("", bootstrapKubeconfigFile)
+				cfg = newBootstrapKubeconfig
+			}
```

Comment:
Maybe we should log a warning when it matches the string, and then also log an error when it doesn't match, e.g. 

`err != nil && strings.Contains(err.Error(), "x509: certificate signed by unknown authority")` == log warning
else if `err !=nil` then log the error

Reply:
changed.

Reply:
FYI: https://github.com/googleapis/gax-go/blob/main/v2/invoke.go#L86-L94